lib/msal-common/src/client/OnBehalfOfClient.ts

```diff
@@ -147,19 +143,49 @@ export class OnBehalfOfClient extends BaseClient {
     }
 
     /**
-     * read account from cache, TODO: CacheManager API should be used here instead
-     * @param account
+     * Fetches the cached access token based on incoming assertion
+     * @param clientId
+     * @param request
+     * @param userAssertionHash
      */
-    private readAccountFromCache(account: AccountInfo): AccountEntity | null {
-        return this.cacheManager.readAccountFromCache(account);
+    private readAccessTokenFromCacheForOBO(clientId: string, request: CommonOnBehalfOfRequest) {
+        const authScheme = request.authenticationScheme || AuthenticationScheme.BEARER;
+        /*
+         * Distinguish between Bearer and PoP/SSH token cache types
+         * Cast to lowercase to handle "bearer" from ADFS
+         */
+        const credentialType = (authScheme && authScheme.toLowerCase() !== AuthenticationScheme.BEARER.toLowerCase()) ? CredentialType.ACCESS_TOKEN_WITH_AUTH_SCHEME : CredentialType.ACCESS_TOKEN;
```

Comment:
is there a reason AuthenticationScheme.BEARER isn't lowercased in the definition and we change the rest of the code base to read 
```ts
(authScheme && authScheme.toLowerCase() !== AuthenticationScheme.Bearer)
```

Reply:
We've unfortunately been a bit inconsistent with casing for enums.